### DISCUSSION TITLE
Any question ?

### DISCUSSION THREAD

**Discussion by Dahkenangnon** [Dahkenangnon/flutter_feathersjs.dart#3] [General]
Welcome at flutter_feathersjs.

Ask here please, any question about how to do thing.

**Comment by kevinbeck76** [comment]
How do you detect errors with socketio methods?   For a call like:


var sessionInfo = await _picServer.scketio.get(serviceName: "sessions", objectId: sessionId);

If the session is found a list is returned with the object JSON in sessionInfo[1].

Session join info: [null, {_id: 5ff9e945189544aabbdc940f, description: , invited_users: [], muted_users: [], active_users: [], name: 7239611028, images_dir: ./data/7239611028, creator_email: daffyduck@looneytunes.com, creator_id: 5ff8c19e4d3f7b8c218e8240, createdAt: 2021-01-09T17:35:01.813Z, updatedAt: 2021-01-09T17:35:01.813Z, __v: 0}]

For invalid objectId values a JsonMap is returned.

{name: NotFound, message: No record found for id '5ff9e945189544aabbdc940e', code: 404, className: not-found, errors: {}}

What is the best way to check for success/failure from these API calls?

Thanks,

Kevin

**Reply by Dahkenangnon** [reply]
Hi @kevinbeck76 

So, for the moment the only  way to know if an error occured in scketio method is:

1. list is returned for successful response (response[1] is exactly what featers js server response)
2. map returned if an error occured

So you should check in what situation you're.


Proper method to handle error will be added soonely for the next major version.

Thank

**Reply by Dahkenangnon** [reply]
This is now fixed in the lastest version: https://pub.dev/packages/flutter_feathersjs
